### PR TITLE
Add tests for AccumuloConfiguration.isPropertySet implementations

### LINKED ISSUE #3486: Add tests for AccumuloConfiguration.isPropertySet
**Is your feature request related to a problem? Please describe.**

Was looking into #3491 and as part of that investigation I was wondering if AccumuloConfiguration.isPropertySet() was behaving properly.  Looked through all of the isPropertySet() implementations and each seemed good.  Looked at the tests and could only find that SystemConfiguration had a test for isPropertySet() and that it only tested the true case and not the false case.

**Describe the solution you'd like**

Adding unit test that verify the expected behavior of AccumuloConfiguration.isPropertySet for each impl would be useful to avoid regressions. Not sure if its workable, but wondering if it would make sense to create generic unit test suite for AccumuloConfiguration impls that can verify certain expected behaviors OR is it easier to have a unit test for each impl.

### PR DESCRIPTION
**Pull request by DomGarguilo**
Closes #3486 

This PR adds a new tests class with a test case for each impl of AccumuloConfiguration to verify that its implementation of `isPropertySet()` is behaving as expected.

### CODE REVIEW

**Review by ctubbsii — CHANGES_REQUESTED** [review]
Found one issue. Didn't have time to finish looking, but the overall approach seems fine. But, please make sure you call EasyMock.verify on the mock objects.

File: server/base/src/test/java/org/apache/accumulo/server/conf/AccumuloConfigurationIsPropertySetTest.java
Comment (keith-turner) [inline server/base/src/test/java/org/apache/accumulo/server/conf/AccumuloConfigurationIsPropertySetTest.java:189]: This needs the same change as mentioned elsewhere

```suggestion
    Set<Property> shouldNotBeSet = Sets.difference(ALL_PROPERTIES, shouldBeSet);
```

File: server/base/src/test/java/org/apache/accumulo/server/conf/AccumuloConfigurationIsPropertySetTest.java
Comment (keith-turner) [inline server/base/src/test/java/org/apache/accumulo/server/conf/AccumuloConfigurationIsPropertySetTest.java:284]: Seems like this set should be the following based on what is set for the parent and table conf object.

```suggestion
    Set<Property> shouldBeSet = Set.of(TABLE_BLOOM_SIZE, TABLE_DURABILITY, TABLE_BLOOM_ENABLED);
```

File: server/base/src/test/java/org/apache/accumulo/server/conf/AccumuloConfigurationIsPropertySetTest.java
Comment (keith-turner) [inline server/base/src/test/java/org/apache/accumulo/server/conf/AccumuloConfigurationIsPropertySetTest.java:285]: Could this do the set difference with all props that other test have done?

**Review by EdColeman — APPROVED** [review]

**Comment by DomGarguilo** [comment]
Some of the tests are failing when i correct the Sets.difference() issue
mentioned in a previous comment. Will work to fix that tomorrow.

**Comment by DomGarguilo** [comment]
Correcting the usage of Sets.difference() in the remaining two test cases (in commit eceaedf) has caused `systemConfiguration()` and `siteConfuguration()` to fail.

The `siteConfuguration()` test fails the check where `getProperties()` is called on the object and many properties are set that are not expected to be. 

The `systemConfiguration` test is failing in a similar way except the `getProperties()` returns the expected properties but the `isPropertySet()` impl is indicating there are properties that are set that are not expected to be.

I am not super familiar with these two and I'm assuming there are some default properties that are included with the set of properties that are manually set upon initializing the object however I don't know where those are coming from or which props are expected to be there.

**Comment by DomGarguilo** [comment]
Fixed the two test cases that were failing in bd3ef97 and bd3ef97. Should be ready for review now.

**Review by ctubbsii — APPROVED** [review]
@DomGarguilo I added a commit to this branch to improve the test. I kept all the test cases the same (well, some of the property values being set changed, but the actual values don't matter; that was just for convenience). But, I reorganized quite a bit of it.

I think it's ready to merge, but I'll leave it to you to go over what I changed, and determine if you agree with the changes I made before you merge it.

**Comment by DomGarguilo** [comment]
> @DomGarguilo I added a commit to this branch to improve the test. I kept all the test cases the same (well, some of the property values being set changed, but the actual values don't matter; that was just for convenience). But, I reorganized quite a bit of it.
> 
> I think it's ready to merge, but I'll leave it to you to go over what I changed, and determine if you agree with the changes I made before you merge it.

The changes you made look great! I'll go ahead and merge this now.

**Pull request merged**
